### PR TITLE
Dependency updates

### PR DESCRIPTION
**Pull request by rubengees**
As the title says.

### CODE REVIEW

File: app/src/main/java/com/vanniktech/emoji/sample/MainActivity.java [hunk 1]
```diff
@@ -30,6 +30,7 @@
 import com.vanniktech.emoji.one.EmojiOneProvider;
 import com.vanniktech.emoji.twitter.TwitterEmojiProvider;
 
+@SuppressWarnings("CPD-START") // We don't care about duplicate code in the sample.
```
Comment (rubengees) [inline app/src/main/java/com/vanniktech/emoji/sample/MainActivity.java]: Found here: https://stackoverflow.com/a/18625921/4279995
Not optimal, but I think there is no better way...

File: build.gradle [hunk 2]
```diff
@@ -17,19 +17,21 @@ buildscript {
...
             ],
         ],
-        'fastLaneScreenGrab': "tools.fastlane:screengrab:1.0.3",
+        'fastLaneScreenGrab': "tools.fastlane:screengrab:1.1.0",
     ]
 
     repositories {
         jcenter()
         google()
+        maven { url "https://plugins.gradle.org/m2/" }
```
Comment (rubengees) [inline build.gradle]: We need this for the `gradle-errorprone-plugin`.

File: emoji/src/test/java/com/vanniktech/emoji/EmojiManagerTest.java [hunk 3]
```diff
@@ -189,16 +189,6 @@ public void noProviderInstalled() {
     assertThat(text.getSpans(0, text.length(), EmojiSpan.class)).hasSize(1);
   }
 
-  @Test public void takeLongest() {
```
Comment (rubengees) [inline emoji/src/test/java/com/vanniktech/emoji/EmojiManagerTest.java]: This was a duplicate of the `halfPath` test.

File: build.gradle (hunk 2 above)
Comment (vanniktech) [inline build.gradle]: We don't. The code quailty plugin takes care of that.

File: build.gradle [hunk 4]
```diff
@@ -4,7 +4,7 @@ apply plugin: 'com.vanniktech.android.junit.jacoco'
 
 buildscript {
     ext.versions = [
-        testRules: '0.5',
+        testRules: '1.0.0',
```
Comment (vanniktech) [inline build.gradle]: i haven't had luck with 1.0.0 and espresso 3.0.0. Did you test locally whether the screenshots are being taken or not. You can do so using `gw assembleDebug assembleAndroidTest && fastlane screengrab`

File: build.gradle [hunk 5]
```diff
@@ -17,19 +17,21 @@ buildscript {
...
+        maven { url "https://plugins.gradle.org/m2/" }
     }
     dependencies {
         classpath 'com.vanniktech:gradle-code-quality-tools-plugin:0.7.0'
-        classpath 'com.vanniktech:gradle-android-junit-jacoco-plugin:0.6.0'
+        classpath 'com.vanniktech:gradle-android-junit-jacoco-plugin:0.7.0'
 
         classpath 'com.android.tools.build:gradle:2.3.3'
         classpath 'com.github.ben-manes:gradle-versions-plugin:0.15.0'
+        classpath 'net.ltgt.gradle:gradle-errorprone-plugin:0.0.11'
```
Comment (vanniktech) [inline build.gradle]: not needed. The code quailty plugin takes care of that.

File: build.gradle (hunk 2 above)
Comment (rubengees) [inline build.gradle]: Oops, I'll revert it.

File: build.gradle (hunk 4 above)
Comment (rubengees) [inline build.gradle]: You are right, at first it did not work. Managed to fix with my latest commit though.

File: build.gradle [hunk 6]
```diff
@@ -4,7 +4,8 @@ apply plugin: 'com.vanniktech.android.junit.jacoco'
 
 buildscript {
     ext.versions = [
-        testRules: '0.5',
+        espresso: '3.0.0',
```
Comment (rubengees) [inline build.gradle]: Concerning screengrab: Something set a different version, so I redefinied it here. Why does that happen?

File: build.gradle (hunk 5 above)
Comment (rubengees) [inline build.gradle]: Fixed.

File: build.gradle (hunk 6 above)
Comment (vanniktech) [inline build.gradle]: Oh it was simply missing here in the first place. Now it should be all good.

**Pull request merged**